compiler/scenario-service/client/src/DA/Daml/LF/ScenarioServiceClient.hs

```diff
@@ -88,7 +88,7 @@ data Handle = Handle
   , hContextId :: IORef LowLevel.ContextId
   -- ^ The root context id, this is mutable so that rather than mutating the context
   -- we can clone it and update the clone which allows us to safely interrupt a context update.
-  , hRunningHandlers :: MVar (MS.Map RunOptions (ThreadId, MVar Bool))
+  , hRunningHandlers :: MVar (MS.Map RunOptions (ThreadId, LowLevel.ContextId, MVar Bool, Barrier (Either LowLevel.Error LowLevel.ScenarioResult)))
```

Comment:
I think it would be worth mentioning why a Barrier is needed/used here.

Reply:
I've opted to split the four-tuple into a RunInfo datatype with comments on the fields